src/dataAccessLayer.ts

```diff
@@ -32,18 +35,46 @@ export class DataAccessLayer {
         return await this.blockchainRepository.getBlockchainAsync();
     }
 
+    public async getBlocksByPeriodAsync(beginTimestamp: number, endTimestamp: number): Promise<Block[]> {
+        return await this.blockchainRepository.getBlocksByPeriodAsync(beginTimestamp, endTimestamp);
+    }
+
+    public async getBlocksByDateAsync(timestamp: number): Promise<Block[]> {
+        return await this.blockchainRepository.getBlocksByDateAsync(timestamp);
+    }
+
+    public async getBlocksByHashAsync(blockHash: string): Promise<Block[]> {
+        return await this.blockchainRepository.getBlocksByHashAsync(blockHash);
+    }
+
     public async getBlockAsync(blockNumber: number): Promise<Block> {
         return await this.blockchainRepository.getBlockAsync(blockNumber);
     }
 
-    public async setBlocksAsync(blocks: Block[]): Promise<void> {
-        await this.blockchainRepository.setBlocksAsync(blocks);
+    public async getPreviousBlockAsync(parentHash: string): Promise<Block> {
+        return await this.blockchainRepository.getPreviousBlockAsync(parentHash);
+    }
+
+    public async getNextBlockAsync(blockHash: string): Promise<Block> {
```

Comment:
It is impossible to get the next block by the current's block hash stored in the header. I dont know if the field of the property should be implemented in this way, and if so... would that implementation be contradicting the core logic of blockchain. @TimoF97  @quintaartsen what are your thoughts?

Reply:
The functionality of `getNextBlockAsync` could be made without having to change the properties of a block by either writing a specific query or an iterative (for-loop) approach. We do indeed not want to change the properties as this would contradict with the core logic of the blockchain.

However, @Juuldamen why would we need this functionality at all?

IS het nodig?

Reply:
I don't know why this function would be needed either so it will be removed. I originally added it based on the endpoint specifications but didn't give it enough thought.

Reply:
After verbal discussion it was decided that this function will be removed.